app/src/test/java/com/android/streetworkapp/ui/profile/SettingsTest.kt

```diff
@@ -0,0 +1,116 @@
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.never
+import org.mockito.Mockito.verify
+import org.mockito.kotlin.whenever
+
+@RunWith(AndroidJUnit4::class)
+class SettingsTest {
+
+  private lateinit var authService: GoogleAuthService
+  private lateinit var userViewModel: UserViewModel
+  private lateinit var parkViewModel: ParkViewModel
+  private lateinit var eventViewModel: EventViewModel
+  private lateinit var progressionViewModel: ProgressionViewModel
+  private lateinit var workoutViewModel: WorkoutViewModel
+  private lateinit var currentUser: User
+
+  @Before
+  fun setUp() {
+    authService = mock(GoogleAuthService::class.java)
+    userViewModel = mock(UserViewModel::class.java)
+    parkViewModel = mock(ParkViewModel::class.java)
+    eventViewModel = mock(EventViewModel::class.java)
+    progressionViewModel = mock(ProgressionViewModel::class.java)
+    workoutViewModel = mock(WorkoutViewModel::class.java)
+    currentUser = User("uid-alice", "Alice", "alice@gmail.com", 42, emptyList(), "")
+  }
+
+  @Test
+  fun testDeleteAccountSuccess() {
+    whenever(userViewModel.currentUser).thenReturn(MutableStateFlow(currentUser))
+    whenever(authService.getCurrentUser()).thenReturn(mock(FirebaseUser::class.java))
+
+    val result =
+        deleteAccount(
+            authService,
+            userViewModel,
+            parkViewModel,
+            eventViewModel,
+            progressionViewModel,
+            workoutViewModel)
+
+    assertTrue(result)
+    verify(authService).deleteAuthUser()
+    verify(userViewModel).deleteUserByUid("uid-alice")
+    verify(userViewModel).removeUserFromAllFriendsLists("uid-alice")
+    verify(parkViewModel).deleteRatingFromAllParks("uid-alice")
+    verify(eventViewModel).removeParticipantFromAllEvents("uid-alice")
+    verify(progressionViewModel).deleteProgressionByUid("uid-alice")
+    verify(workoutViewModel).deleteWorkoutDataByUid("uid-alice")
+  }
+
+  @Test
+  fun testDeleteAccountFailsDueToEmptyUid() {
+    whenever(userViewModel.currentUser).thenReturn(MutableStateFlow(currentUser))
+
+    val result =
+        deleteAccount(
+            authService,
+            userViewModel,
+            parkViewModel,
+            eventViewModel,
+            progressionViewModel,
+            workoutViewModel)
+
+    assertFalse(result)
+    verify(authService, never()).deleteAuthUser()
+    verify(userViewModel, never()).deleteUserByUid(anyString())
+    verify(userViewModel, never()).removeUserFromAllFriendsLists(anyString())
+    verify(parkViewModel, never()).deleteRatingFromAllParks(anyString())
+    verify(eventViewModel, never()).removeParticipantFromAllEvents(anyString())
+    verify(progressionViewModel, never()).deleteProgressionByUid(anyString())
+    verify(workoutViewModel, never()).deleteWorkoutDataByUid(anyString())
+  }
+
+  @Test
+  fun testDeleteAccountFailsDueToNullFirebaseUser() {
+    whenever(userViewModel.currentUser).thenReturn(MutableStateFlow(currentUser))
```

Comment:
This call could be put in the setUp instead of repeating it in every test

Reply:
Good suggestion, I've made the suggested change.